### PR TITLE
feat: Fix/Enhance Routes; Add uid workaround for repo server

### PR DESCRIPTION
**Pull request by tylerauerbeck**
Checklist:

* [ X] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [ X] Any new values are backwards compatible and/or have sensible default.
* [ X] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [X ] I have signed the CLA and the build is green.
* [X ] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

This PR:
- Modifies routes so that they can be enabled as passthrough/edge and also allows for modification of insecure redirect (http -> https). Also adds an empty status field in order to get helm3 to quit complaining.
- Bumps dex versions to v2.22.0 to provide OpenShift connector 
- Adds an openshift variable. This could allow for a number of things in the future (enable/configure authentication, etc.), but is currently in place so that the uid_entrypoint.sh workaround can be enabled in the repo-server deployment (which allows configuring ssh/git configurations in OpenShift)

### CODE REVIEW

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml [hunk 1]
```diff
@@ -64,7 +67,14 @@ spec:
         {{- if .Values.repoServer.containerSecurityContext }}
         securityContext: {{- toYaml .Values.repoServer.containerSecurityContext | nindent 10 }}
         {{- end }}
-        {{- if .Values.repoServer.env }}
+{{- if .Values.openshift.enabled }}
+        env:
+        - name: USER_NAME
+          value: argocd
+{{- if .Values.repoServer.env }}
```
Comment (spencergilbert) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml]: Can we combine the two repoServer.env templates here?

**Comment by spencergilbert** [comment]
I'm not familiar enough with openshift to really troubleshoot/understand the changes you've made, left one comment aside from that @tylerauerbeck

**Comment by spencergilbert** [comment]
@tylerauerbeck can you look at the check failures?

**Comment by tylerauerbeck** [comment]
@spencergilbert Taking a look at this today. Will get your comments and merge conflicts cleaned up.

**Comment by spencergilbert** [comment]
Sounds good - just @ me when it's ready. I'm busy but I'll be sure to check in tonight

**Comment by tylerauerbeck** [comment]
@spencergilbert All looks good now. Let me know if you have any other concerns.

**Comment by yann-soubeyrand** [comment]
@tylerauerbeck What's the reason why this PR reintroduces the status field? This prevents Argo CD from converging in applying this chart.

**Pull request merged**